### PR TITLE
SIR-1522 - cli v2 pre flight validation gate on raisely deploy no validate

### PR DESCRIPTION
**Pull request by KeinerM**
## What changed
- Added a deploy pre-flight validation gate for campaign SCSS and component source.
- Added `--no-validate` on `raisely deploy` to bypass validation when needed.
- Deploy now exits before uploads when validation fails, and prints all failures with campaign/component context.
- Added integration-style deploy tests covering:
  - bad SCSS validation
  - bad component validation
  - network failure during validation
  - `--no-validate` bypass
  - validation gating under `--force`
  - validation gating when `cli: true`
- Migrated the CLI test suite from `node:test` to `vitest`.

---

> [!NOTE]
> **High Risk**
> Deploy now performs pre-upload SCSS/component validation and includes token-refreshing validator calls, plus new test tooling/deps; mistakes here could block deployments or mishandle auth/validation flows.
> 
> **Overview**
> Adds a **pre-flight validation gate** to `raisely deploy` that validates campaign SCSS (via remote transpiler) and local component source, failing fast *before any uploads* and printing all errors with campaign/component context. Introduces `--no-validate` to bypass this gate when needed (validation still runs under `--force`/`cli: true`).
> 
> Adds new validator helpers (`validateCampaignSass`, `validateComponent`) including transpiler URL normalization, credential refresh-on-401, and graceful handling of malformed/failed validator responses, plus deploy-focused tests covering validation failures/bypass and network/auth edge cases. Migrates the test suite to `vitest` and updates test config/scripts.
> 
> ## Risk Assessment (Framework v2.0)
> 
> | Dimension | Tier | Score | Rationale |
> |---|---|---|---|
> | Security | 1 | Low | No findings |
> | Sensitive domains | 1 | Medium | Touches token/credential refresh paths during validation (`validateCampaignSass`) |
> | Data integrity | 1 | Low | No data access changes |
> | Deployment | 1 | Medium | Changes deploy command behavior and adds build/test tooling (`vitest`) |
> | Scope | 2 | Medium | Moderate modifications to `deploy` flow plus new tests |
> | Architecture | 2 | Low | Changes are contained within CLI command/actions/tests |
> | Feature flags | 2 | Low | Internal CLI/tooling change; flags not applicable |
> | Dependencies | 2 | High | Adds `vitest` dev dependency and updates test script/config |
> 
> ### Mitigating Factors
> 
> - Adds targeted deploy/validation test coverage for failure and bypass scenarios.
> 
> ### Findings
> 
> | # | Finding | Status | Resolution |
> |---|---|---|---|
> | 1 | Validation now blocks deploy uploads; ensure error messaging and `--no-validate` behavior are acceptable for all workflows (CI and interactive). | Open | -- |
> 
> **Outstanding findings: 1**
> **Overall risk: HIGH**
> **Decision: ESCALATE**
> **Rationale:** Two Tier 1 dimensions are Medium (auth-adjacent validation + deploy behavior change), and there is an open workflow-risk finding that warrants human review.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 4b76e0cf1841ba65c52c66c641017e45a0402cf2. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: src/deploy.js [hunk 1]
```diff
@@ -21,35 +21,139 @@ import {
...
-	await getToken(program, config);
-
-	welcome();
-	log(`You are about to deploy your local directly to Raisely`, 'white');
-	br();
-	console.log(`    ${chalk.inverse(`${process.cwd()}`)}`);
-	br();
+	let config = await deps.loadConfigFn();
+	config.token = await deps.getTokenFn(program, config);
+
```
[the remark this answers is not in the document]
Comment (KeinerM) [reply src/deploy.js]: The issue does not occur in the current implementation.
`deploy` waits for authentication first:
- `config.token = await deps.getTokenFn(program, config)`
Then it runs validation with that same `config` object:
- `runPreflightValidation({ config }, deps)`
Inside validation, the token used is `config.token`, so pre-flight validation runs only after token retrieval completes.

File: src/deploy.js [hunk 2]
```diff
@@ -21,35 +21,169 @@ import {
...
+					'Component validator returned an invalid response.'
+				);
+				return {
+					ok: result.ok,
+					context: `Component ${name}`,
+					error: result.error,
+				};
+			})
+		),
+	];
```
[the remark this answers is not in the document]
Comment (KeinerM) [reply src/deploy.js:130]: The loader status is updated only after `await Promise.all(validationTasks)`.

So both `Validation passed` and `Deploy validation failed` are emitted only after all validation tasks complete, including later batches under the concurrency limit.

File: src/deploy.js [hunk 3]
```diff
@@ -61,23 +195,36 @@ export default async function deploy(options = {}) {
...
 		if (!response.confirm) {
-			br();
-			return log('Deploy aborted', 'red');
+			deps.brFn();
+			return deps.logFn('Deploy aborted', 'red');
 		}
 	}
 
+	if (options.validate !== false) {
+		const validationPassed = await runPreflightValidation({ config }, deps);
```
[the remark this answers is not in the document]
Comment (KeinerM) [reply src/deploy.js]: In this flow, `--no-validate` only skips the validation gate; deploy still uploads styles/components/pages, which requires authentication.
So token retrieval before that step is still required. This is not an unnecessary auth call caused by `--no-validate`.

File: package.json [hunk 4]
```diff
@@ -40,7 +40,8 @@
 		"p-limit": "^4.0.0"
 	},
 	"devDependencies": {
-		"prettier": "^2.6.2"
+		"prettier": "^2.6.2",
+		"vitest": "^1.6.1"
```
Comment (xolott-ark) [inline package.json]: Can you pin this dependency? We should pin the others as well :(

**Review by xolott-ark — CHANGES_REQUESTED** [review]
One last change here 🙏

**Review by xolott-ark — APPROVED** [review]
